### DISCUSSION TITLE
GitLab Neuron Template

### DISCUSSION THREAD

**Discussion by RoyiAvital** [srid/neuron#506] [Ideas]
Since there is no support for Windows (Native) for Neuron the only way to use it is GitHub Neuron Template.
The problem is the website is public even if the repository is private.

What I'd like to suggest is:

1. Build a template for GitLab which offers static pages even from private repositories and with limited access to the sites.
2. Allow a mode of the template to generate the HTML files and push them into the repository without publishing a web site. Namely only as an engine to generate the HTML files in the repository.

**Comment by srid** [comment]
> Build a template for GitLab which offers static pages even from private repositories and with limited access to the sites.

I don't use Gitlab, but [the documentation](https://neuron.zettel.page/778816d3.html) points to https://gitlab.com/thematten/neuron-template -- have you tried that?

CC @TheMatten

**Comment by RoyiAvital** [comment]
For some reason I have missed that.
Still, any way to make it generate the HTML pages and push into the repository (So it will be available to pull)?

By the way, I think you need to has a flag in the options for the meta flag of robots.
For instance, add option called `blockWebCrawlers` which, if set to `true` will add at the header:

```html
<meta name="robots" content="noindex">
```

See [Block Search Indexing with 'noindex'](https://developers.google.com/search/docs/advanced/crawling/block-indexing).

**Comment by srid** [comment]
> Still, any way to make it generate the HTML pages and push into the repository (So it will be available to pull)?

You can try this action: https://github.com/marketplace/actions/git-auto-commit

Custom `<head>` elements can be put in `head.html`, see https://neuron.zettel.page/custom-head.html

**Comment by RoyiAvital** [comment]
The template doesn't have the `head` file. Of course I'd like to start form the default.
Moreover, this should be an option for beginners as well. They are ok with the defaults, just need this simple feature.

It shouldn't be hard to add it, right?

**Comment by srid** [comment]
Yes, you can just add `head.html` at top-level and neuron will pick it up (regardless where your notes are hosted, github or gitlab).

**Comment by RoyiAvital** [comment]
Yes. But I don't want to mess with the current `head`. I just want to add to it a single line.
If the default `head` was part of the files in template, I could edit it.
But the way it is, I either have to recreate everything or not have what I need.

What I want is the current head (I like the design) and be able to add lines to the `head` section.
